utils/utils.go

```diff
@@ -25,6 +26,9 @@ func DownloadWithProgress(url string, tarName string, destFolder string) (err er
 	if err != nil {
 		return err
 	}
+	if resp.StatusCode == http.StatusNotFound {
+		return fmt.Errorf("%s 404 not found", url)
+	}
```

Comment:
What about it, if we have another error code?

For example, if we have 5xx code, error during delivering body.

Maybe check like this?

```golang
if resp.StatusCode != http.StatusOK {
	return fmt.Errorf("%d code in in processing url: %s", resp.StatusCode, url)
}
```

Reply:
Thanks